### PR TITLE
[AUS] expose metrics for addon upgrades

### PR DESCRIPTION
**Pull request by geoberle**
expose metrics for addon upgrades

* `aus_addon_upgrade_policy_info` - cluster and addon metadata, including upgrade policy details
* `aus_addon_version_remaining_soak_days` - remaining soak days for a addon version on a cluster

they have the same set of labels as their cluster upgrade counterparts but have an additional one called `addon` to distinguish between addons managed on the same cluster.

part of [APPSRE-8524](https://issues.redhat.com/browse/APPSRE-8524)

### CODE REVIEW

File: reconcile/aus/ocm_upgrade_scheduler.py [hunk 1]
```diff
@@ -63,26 +54,21 @@ def process_upgrade_policies_in_org(
...
 
-    def get_ocm_env_upgrade_specs(
-        self, ocm_env: OCMEnvironment
-    ) -> dict[str, OrganizationUpgradeSpec]:
-        raise NotImplementedError(
-            "Don't use ocm-upgrade-scheduler anymore but use: \n"
-            "* ocm-label to transfer upgrade policies to OCM subscription labels \n"
-            "* advanced-upgrade-service to drive upgrade policies based on OCM subscription labels"
-        )
-
```
Comment (geoberle) [inline reconcile/aus/ocm_upgrade_scheduler.py:85]: for the reviewer. unrelated cleanup i snuck in

File: reconcile/aus/ocm_upgrade_scheduler.py [hunk 2]
```diff
@@ -63,26 +54,21 @@ def process_upgrade_policies_in_org(
             version_data=version_data,
         )
         self.expose_remaining_soak_day_metrics(
-            ocm_env=org_upgrade_spec.org.environment.name,
             org_upgrade_spec=org_upgrade_spec,
             version_data=version_data,
             current_state=current_state,
+            metrics_builder=functools.partial(
```
Comment (patjlm) [inline reconcile/aus/ocm_upgrade_scheduler.py:60]: TIL `functools.partial`

**Review by patjlm — APPROVED** [review]
nice!
